src/Control/Effect/Resumable.hs

```diff
@@ -22,12 +22,16 @@ import Control.Monad.Trans.Class
 import Data.Functor.Classes
 
 -- | Errors which can be resumed with values of some existentially-quantified type.
-data Resumable err (m :: * -> *) k
-  = forall a . Resumable (err a) (a -> k)
+data Resumable err m k
+  = forall a . Resumable (err a) (a -> m k)
 
-deriving instance Functor (Resumable err m)
-deriving instance HFunctor (Resumable err)
-deriving instance Effect (Resumable err)
+deriving instance Functor m => Functor (Resumable err m)
+
+instance HFunctor (Resumable err) where
+  hmap f (Resumable err k) = Resumable err (f . k)
+
+instance Effect (Resumable err) where
+  handle state handler (Resumable err k) = Resumable err (handler . (<$ state) . k)
```

Comment:
We can no longer derive these two instances, because deriving them requires a `Generic1` instance, which itself can’t be derived for `Resumable` due to the existential quantification.